### PR TITLE
feat(hooks): implement ClearRefinements component

### PR DESCRIPTION
**Pull request by sarahdayan**
## Summary

This introduces the `ClearRefinements` DOM widget.

Since we're moving away from the `translations` and `templates` APIs, I introduced a `resetLabel` prop to pass a custom label (after agreeing on this with the team). It reuses the [same name as in InstantSearch.js](https://www.algolia.com/doc/api-reference/widgets/clear-refinements/js/#widget-param-resetlabel), thus makes the APIs closer.

We can do a pass once all widgets are implemented to revise whether we're happy with those APIs and harmonize them if needed.

Ticket: [FX-1150](https://algolia.atlassian.net/browse/FX-1150)

## Results

- New `ClearRefinements` widget
- DOM tests for both UI component and widget
- Replaced `ClearRefinements` with the widget in example

### CODE REVIEW

**Comment by sarahdayan** [comment]
The failing tests are unrelated to the new widget, and it seems they aren't what affects them (running tests on `feat/hooks-dom-hits` also fails locally, using the same Node version as the CI—I don't know what makes them pass on #3363).

Since tests are passing for the widget, the implementation can be reviewed now while I find out why there's a discrepancy in unit test output between both PRs.

File: packages/react-instantsearch-hooks-dom/src/index.ts
Comment (Haroenv) [inline packages/react-instantsearch-hooks-dom/src/index.ts:3]: maybe add Hits to the barrel in this PR too then

**Review by Haroenv — COMMENTED** [review]
👌

**Comment by Haroenv** [comment]
I'm not sure why, but the answers test seems flaky, I've seen it pass and fail without any code changes. Maybe time to try fixing it in a separate PR?

File: packages/react-instantsearch-hooks-dom/package.json
Comment (sarahdayan) [reply packages/react-instantsearch-hooks-dom/package.json:58]: Yep, will rebase.

File: packages/react-instantsearch-hooks-dom/src/index.ts
Comment (sarahdayan) [reply packages/react-instantsearch-hooks-dom/src/index.ts:3]: I wasn't sure you were going with it, but fine by me.

File: packages/react-instantsearch-hooks-dom/src/ui/ClearRefinements.tsx
Comment (sarahdayan) [reply packages/react-instantsearch-hooks-dom/src/ui/ClearRefinements.tsx:14]: Agreed, c.f. [PR description](https://github.com/algolia/react-instantsearch/pull/3367#issue-1148258249). The RFC doesn't cover whether `templates` should be standardized or not, but let's discuss this once we're done with all widgets.

File: packages/react-instantsearch-hooks-dom/src/widgets/__tests__/ClearRefinements.test.tsx
Comment (sarahdayan) [reply packages/react-instantsearch-hooks-dom/src/widgets/__tests__/ClearRefinements.test.tsx:34]: React is complaining that changes aren't wrapped in an `act` because InstantSearch first renders with default values, then updates the UI. We're using this workaround in the `DynamicWidgets` component as well (c.f. internal conversation from yesterday on React tests).

**Comment by sarahdayan** [comment]
> I'm not sure why, but the answers test seems flaky, I've seen it pass and fail without any code changes. Maybe time to try fixing it in a separate PR?

Yep, see [my comment](https://github.com/algolia/react-instantsearch/pull/3367#issuecomment-1048947292).

File: packages/react-instantsearch-hooks-dom/src/widgets/ClearRefinements.tsx
Comment (sarahdayan) [inline packages/react-instantsearch-hooks-dom/src/widgets/ClearRefinements.tsx:30]: @Haroenv Realized I forgot this. This isn't easy to test at this level without checking internals.

File: packages/react-instantsearch-hooks-dom/package.json
Comment (sarahdayan) [reply packages/react-instantsearch-hooks-dom/package.json]: Addressed in 201dccc2.

File: packages/react-instantsearch-hooks-dom/src/index.ts
Comment (sarahdayan) [reply packages/react-instantsearch-hooks-dom/src/index.ts]: Addressed in 45324117.

**Comment by sarahdayan** [comment]
@Haroenv Let's fix the flaky test and add the `$$widgetType` testing for separate PRs since this will benefits all components and not just this one, fine by you?

**Review by dhayab — COMMENTED** [review]
Looks great!

I'm guessing you're waiting until #3363 is merged? If so, I'll do another check once its done and this PR is rebased.

File: packages/react-instantsearch-hooks-dom/src/widgets/__tests__/ClearRefinements.test.tsx
Comment (Haroenv) [reply packages/react-instantsearch-hooks-dom/src/widgets/__tests__/ClearRefinements.test.tsx:34]: I noticed these warnings now as well, and I think it might be better to leave them for now and find out why they exist, if we can avoid it somehow (likely related to useEffect+widget mount on render) than ignoring them now already? maybe https://twitter.com/acdlite/status/1496175720971874308 is related?

**Comment by Haroenv** [comment]
once rebased lgtm :)

File: packages/react-instantsearch-hooks-dom/src/widgets/__tests__/ClearRefinements.test.tsx
Comment (sarahdayan) [reply packages/react-instantsearch-hooks-dom/src/widgets/__tests__/ClearRefinements.test.tsx:34]: Yes it makes sense. I'll remove this for now since they "only" clutter the output but don't fail tests and we can take global action for the whole suite in the end.

File: packages/react-instantsearch-hooks-dom/package.json
Comment (Haroenv) [inline packages/react-instantsearch-hooks-dom/package.json:50]: this needs to be updated still, I assume rebase incomplete?

**Review by Haroenv — APPROVED** [review]

**Pull request merged**
